Upstream-Status: Pending
From b66aba3c8fc50e70a67f63e52b54ae687b90b45c Mon Sep 17 00:00:00 2001
From: Rex Lu <rex.lu@mediatek.com>
Date: Mon, 10 Nov 2025 14:28:57 +0800
Subject: [PATCH] test

---
 flowtable.sh | 2 +-
 smp-mt76.sh  | 9 +++++----
 2 files changed, 6 insertions(+), 5 deletions(-)

diff -Naur sources.origin/flowtable.sh sources/flowtable.sh
--- sources.origin/flowtable.sh	2026-08-07 13:38:23.772511155 +0000
+++ sources/flowtable.sh	2026-08-07 13:41:32.082385630 +0000
@@ -38,7 +38,7 @@
 	fi
 
 	# check all of the ETH/WiFi virtual interfaces
-	eth_ifs=$(ls /sys/class/net | grep -E '^(eth|lan|wan|mxl|ppp)')
+	eth_ifs=$(ls /sys/class/net | grep -E '^(eth|lan|wan|mxl|ppp|erouter)')
 	wifi_ifs=$(iw dev | grep 'Interface' | awk '{print $2}')
 
 	interfaces=$(echo -e "$eth_ifs\n$wifi_ifs")
diff -Naur sources.origin/smp-mt76.sh sources/smp-mt76.sh
--- sources.origin/smp-mt76.sh	2026-08-07 13:38:23.772511155 +0000
+++ sources/smp-mt76.sh	2026-08-07 13:44:43.480292830 +0000
@@ -1,13 +1,14 @@
 #!/bin/sh
 
-source /sbin/flowtable.sh
+source /usr/sbin/flowtable.sh
 
 OPTIMIZED_FOR="$1"
 CPU_LIST=`cat /proc/interrupts | sed -n '1p'`
 NUM_OF_CPU=0; for i in $CPU_LIST; do NUM_OF_CPU=`expr $NUM_OF_CPU + 1`; done;
 DEFAULT_RPS=0
 
-. /lib/functions.sh
+#. /lib/functions.sh
+board_name=`cat /proc/device-tree/compatible`
 
 RPS_IF_LIST=""
 NUM_WIFI_CARD=0
@@ -27,7 +28,7 @@
 	NET_IF_LIST=`ls /sys/class/net`
 	for vif in $NET_IF_LIST;
 	do
-		if [[ "$vif" == "eth"* ]] ||  \
+		if [[ "$vif" == "eth"* ]] || [[ "$vif" == "erouter"* ]] || \
 		[[ "$vif" == "lan"* ]] || [[ "$vif" == "wan"* ]] || \
 		[[ "$vif" == "wlan"* ]] || [[ "$vif" == "phy"* ]] || \
 		[[ "$vif" == *"mld"* ]]; then
@@ -353,7 +354,7 @@
 
 setup_model()
 {
-	board=$(board_name)
+	board=$board_name
 	num_of_wifi=$NUM_WIFI_CARD
 
 	if [[ $board == *"7988"* ]]; then
